File: quark/test/emit/Overload/py/Overload.py
from quark_runtime import *

class Overload(object):
    def _init(self):
        self.name = None

    def __init__(self, name):
        self._init()
        (self).name = name

    def __add__(self, o):
        return o

    def __mul__(self, o):
        return self

    def test(self):
        o1 = Overload(u"one");
        o2 = Overload(u"two");
        o3 = (o1).__add__(o2);
        _println((o3).name);
        o3 = (o1).__mul__(o2)
        _println((o3).name);
        if ((o3) == (o1)):
            _println(u"YAY!");

    


def main():
    o = Overload(u"test");
    (o).test();


if __name__ == "__main__":
    main()

File: quark/test/emit/Overload/js/Overload.js
var _qrt = require("datawire-quark-core");

// CLASS Overload

function Overload(name) {
    this.__init_fields__();
    (this).name = name;
}
exports.Overload = Overload;

function Overload__init_fields__() {
    this.name = null;
}
Overload.prototype.__init_fields__ = Overload__init_fields__;

function Overload___add__(o) {
    return o;
}
Overload.prototype.__add__ = Overload___add__;

function Overload___mul__(o) {
    return this;
}
Overload.prototype.__mul__ = Overload___mul__;

function Overload_test() {
    var o1 = new Overload("one");
    var o2 = new Overload("two");
    var o3 = (o1).__add__(o2);
    _qrt.print((o3).name);
    o3 = (o1).__mul__(o2);
    _qrt.print((o3).name);
    if ((o3) === (o1)) {
        _qrt.print("YAY!");
    }
}
Overload.prototype.test = Overload_test;

function main() {
    var o = new Overload("test");
    (o).test();
}
exports.main = main;

main();
